core/dbt/dataclass_schema.py

```diff
@@ -93,7 +90,7 @@ def __pre_deserialize__(cls, data, options=None):
     def _local_to_dict(self, **kwargs):
         args = {}
         if 'omit_none' in kwargs and kwargs['omit_none'] is False:
-            args['options'] = {'keep_none': True}
+            args['omit_none'] = False
         return self.to_dict(**args)
```

Comment:
Is this unnecessary now that we went back to the `omit_none` argument?

Reply:
It's still necessary because it's called from Hologram, which can provide the 'validate' parameters, which is not valid for Mashumaro. But I've updated to pass it along for any value of 'omit_none'.